crates/shared/src/price_estimation/native.rs

```diff
@@ -25,6 +26,30 @@ pub fn default_amount_to_estimate_native_prices_with(chain_id: u64) -> Option<U2
     }
 }
 
+/// Convert from normalized price to floating point price
+pub fn from_normalized_price(price: BigDecimal) -> Option<f64> {
+    // Convert U256 to f64
+    let price_in_eth = price.to_f64()?;
+
+    // Divide by 1e18 to reverse the multiplication by 1e18
+    let normalized_price = price_in_eth / 1e18;
```

Comment:
Can we do the division while using `BigDecimal` and only convert afterwards to avoid losing precision?